strapi/strapi
# Fix Update user does not update component attribute

Signed-off-by: harimkims <harimkims@gmail.com>



### What does it do?

- Replace old update query used in v3 with new `entityService` in v4
- Add e2e test for GraphQL update user mutation

### Why is it needed?

To be compatible with v4 params

### How to test it?

1. Create component
2. Add component field in User in Content-type builder. (single component)
3. Go to page for Create or Update user
4. Input any data into component field
5. Send GraphQL mutation in http://localhost:1337/graphql
```gql
mutation updateUser {
  updateUsersPermissionsUser(id: "3", data: { comp: { id: "1", name: "update test" } }) {
    data {
      id
      attributes {
        comp {
          id
          name
        }
      }
    }
  }
}
```
6. Check if field is updated in result

### Related issue(s)/PR(s)

Closes #11832
